src/rules/ResourceICUPlurals.js

```diff
@@ -66,98 +66,200 @@ class ResourceICUPlurals extends ResourceRule {
-                const missing = neededCategories.filter(category => {
-                    return typeof(opts[category]) === 'undefined';
-                });
-                if ( missing && missing.length ) {
-                    let opts = {
-                        severity: "error",
-                        rule: this,
-                        description: `Missing categories in target string: ${missing.join(", ")}. Expecting these: ${neededCategories.join(", ")}`,
-                        id: key,
-                        highlight: `Target: ${stringToCheck}<e0></e0>`,
-                        pathName,
-                        source,
-                        locale
-                    };
-                    value.push(new Result(opts));
-                }
-                for (let category in opts) {
-                    if ( opts[category] && Array.isArray(opts[category].value) ) {
-                        value = value.concat(this.checkPluralCategories(opts[category].value, neededCategories, stringToCheck, key, pathName, source, locale));
-                    }
-                }
-                // now check the other way around. That is, if the categories that exist are not needed.
-                const extras = Object.keys(opts).filter(category => {
-                    return neededCategories.indexOf(category) < 0;
-                });
-                if (extras && extras.length) {
-                    let opts = {
-                        severity: "warning",
-                        rule: this,
-                        description: `Extra categories in target string: ${extras.join(", ")}. Expecting only these: ${neededCategories.join(", ")}`,
-                        id: key,
-                        highlight: `Target: ${stringToCheck}<e0></e0>`,
-                        pathName,
-                        source,
-                        locale
-                    };
-                    value.push(new Result(opts));
+    matchCategories(sourceSelect, targetSelect, locale, resource) {
+        let problems = [];
+        const srcLocale = new Locale(resource.getSourceLocale());
+
+        // first check the required plural categories
+        let requiredSourceCategories, requiredTargetCategories;
+        let actualRequiredSourceCategories = [], actualRequiredTargetCategories = [];
+        let actualNonrequiredSourceCategories = [], actualNonrequiredTargetCategories = [];
+
+        if (sourceSelect.node.pluralType === "cardinal") {
+            requiredSourceCategories = categoriesForLang[srcLocale.getLanguage()] || [ "one", "other" ]
+            requiredTargetCategories = categoriesForLang[locale.getLanguage()] || [ "one", "other" ];
+        } else {
+            // for select or selectordinal, only the "other" category is required
+            requiredSourceCategories = [ "other" ];
+            requiredTargetCategories = [ "other" ];
+        }
+
+        for (let category in sourceSelect.node.options) {
```

Comment:
for-in is fragile, could we use this instead?
```js
for (const category of Object.keys(sourceSelect.node.options)) {
}
```

---

Or - this whole section could be rewritten as
```js
const sourceCategories = Object.keys(sourceSelect.node.options);
const actualRequiredSourceCategories = sourceCategories.filter(category => requiredSourceCategories.includes(category));
// etc
```

I personally kind of prefer it this way, because it's easier to see at a glance where does the content come from: You first do some stuff to obtain the values (i.e. filter another array), then declare them under a new variable.